### PR TITLE
Sticker Sending

### PR DESCRIPTION
**Pull request by DonovanDMC**
Through inspecting client network requests, and testing with bots, sending `sticker_ids` allows bots to send stickers (multiple as well).

[postman request](https://butts-are.cool/DiscordCanary_07-28-2021_05-32-41.png)

### CODE REVIEW

**Comment by DonovanDMC** [comment]
Bots only seem to be able to send stickers from the current server, so sending in direct messages is basically impossible

**Comment by DonovanDMC** [comment]
Didn't think this was in the docs, but they are


Also, since the max is explicitly 3, we could do `[string?, string?, string?]` for between 0 and 3, but that might not go with the general types style around here

File: index.d.ts [hunk 1]
```diff
@@ -2300,9 +2301,10 @@ declare namespace Eris {
     reactions: { [s: string]: { count: number; me: boolean } };
     referencedMessage?: Message | null;
     roleMentions: string[];
+    stickerItems?: StickerItems[];
     /** @deprecated */
     stickers?: Sticker[];
-    stickerItems?: StickerItems[];
+
```
Comment (conorwastakenwastaken) [inline index.d.ts]: Random newline here 👀

**Review by conorwastakenwastaken — CHANGES_REQUESTED** [review]
Also, is there a reason the property is called `stickerIDS`? Other properties (such as `channelIDs`, `guildIDs`, etc.) don't have a capital S at the end, so I'm just curious if there was a reason tbh

**Comment by DonovanDMC** [comment]
> Also, is there a reason the property is called `stickerIDS`? Other properties (such as `channelIDs`, `guildIDs`, etc.) don't have a capital S at the end, so I'm just curious if there was a reason tbh

the reason is I didn't realize they were like that

File: lib/Client.js [hunk 2]
```diff
@@ -588,6 +588,7 @@ class Client extends EventEmitter {
     * @arg {String} [content.messageReference.guildID] The guild ID of the referenced message
     * @arg {String} content.messageReference.messageID The message ID of the referenced message. This cannot reference a system message
     * @arg {String} [content.messageReferenceID] [DEPRECATED] The ID of the message should be replied to. Use `messageReference` instead
+    * @arg {Array<String>} [content.stickerIDs] An array of ids corresponding to stickers to send
```
Comment (conorwastakenwastaken) [inline lib/Client.js]: I didn't notice this earlier, but for consistency reasons, should `ids` be changed to `IDs` here? And the same for the JS docs in PrivateChannel & TextChannel.

File: index.d.ts [hunk 3]
```diff
@@ -2300,9 +2301,9 @@ declare namespace Eris {
     reactions: { [s: string]: { count: number; me: boolean } };
     referencedMessage?: Message | null;
     roleMentions: string[];
+    stickerItems?: StickerItems[];
```
Comment (conorwastakenwastaken) [inline index.d.ts:2368]: Idk how I didn't notice this was out of order when I made the PR for stickerItems, but I believe it's also alphabetized wrong in the Message structure too.

**Review by conorwastakenwastaken — CHANGES_REQUESTED** [review]

**Comment by conorwastakenwastaken** [comment]
> > Also, is there a reason the property is called `stickerIDS`? Other properties (such as `channelIDs`, `guildIDs`, etc.) don't have a capital S at the end, so I'm just curious if there was a reason tbh
> 
> the reason is I didn't realize they were like that

Ah, gotcha lol

File: lib/Client.js [hunk 4]
```diff
@@ -603,8 +603,8 @@ class Client extends EventEmitter {
                 };
             } else if(content.content !== undefined && typeof content.content !== "string") {
                 content.content = "" + content.content;
-            } else if(content.content === undefined && !content.embed && !content.embeds && !file) {
-                return Promise.reject(new Error("No content, file, or embeds"));
+            } else if(content.content === undefined && !content.embed && !content.embeds && !file && (content.stickerIDS === undefined || content.stickerIDS.length === 0)) {
```
Comment (smhman) [inline lib/Client.js]: Why do you check twice that if all the others + stickers exists? Couldn't you just merge sticker check into the same if closure?

Comment (JustCat80) [inline lib/Client.js]: sir the red highlighted was removed and the green was added *lol the review I replied to was deleted*

**Comment by conorwastakenwastaken** [comment]
Was 951bac5 meant to include a ton of spaces being added to the `if`/`catch` blocks?

**Comment by DonovanDMC** [comment]
for future reference, in case anyone asks about it, it seems like stickers can't be sent via webhooks at all, since it gives an `Cannot send an empty message` error

File: index.d.ts [hunk 5]
```diff
@@ -852,6 +839,21 @@ declare namespace Eris {
     command: Command;
     timeout: NodeJS.Timer;
   }
+
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
```

File: lib/Client.js (hunk 2 above)
Comment (bsian03) [reply lib/Client.js]:
```suggestion
    * @arg {Array<String>} [content.stickerIDs] An array of IDs corresponding to stickers to send
```

File: lib/structures/PrivateChannel.js [hunk 6]
```diff
@@ -72,6 +72,7 @@ class PrivateChannel extends Channel {
     * @arg {String} [content.messageReference.guildID] The guild ID of the referenced message
     * @arg {String} content.messageReference.messageID The message ID of the referenced message. This cannot reference a system message
     * @arg {String} [content.messageReferenceID] [DEPRECATED] The ID of the message should be replied to. Use `messageReference` instead
+    * @arg {Array<String>} [content.stickerIDs] An array of ids corresponding to the stickers to send
```
Comment (bsian03) [inline lib/structures/PrivateChannel.js]:
```suggestion
    * @arg {Array<String>} [content.stickerIDs] An array of IDs corresponding to the stickers to send
```

File: lib/structures/TextChannel.js [hunk 7]
```diff
@@ -93,6 +93,7 @@ class TextChannel extends GuildChannel {
     * @arg {String} [content.messageReference.guildID] The guild ID of the referenced message
     * @arg {String} content.messageReference.messageID The message ID of the referenced message. This cannot reference a system message
     * @arg {String} [content.messageReferenceID] [DEPRECATED] The ID of the message should be replied to. Use `messageReference` instead
+    * @arg {Array<String>} [content.stickerIDs] An array of ids corresponding to the stickers to send
```
Comment (bsian03) [inline lib/structures/TextChannel.js]:
```suggestion
    * @arg {Array<String>} [content.stickerIDs] An array of IDs corresponding to the stickers to send
```

**Review by bsian03 — CHANGES_REQUESTED** [review]
noob

**Review by bsian03 — APPROVED** [review]

**Pull request merged**